### ISSUE TITLE
Error: Debug Failure. Invalid cast. The supplied value [object Object] did not pass the test 'isOptionalChain'.

### ISSUE THREAD

**Issue by Hacksign** [acrazing/typescript-loadable-components-plugin#4]
When I use this plugin with ts-loader and webpack, I got this error:

```
>> npm run startblog

> frontend@1.0.0 startblog /home/hacksign/Code/blogstation
> cross-env NODE_ENV=development node -r ts-node/register ./frontend/index.ts

Example app listening on port 3000!

/home/hacksign/Code/blogstation/node_modules/typescript/lib/typescript.js:1693
        return ts.Debug.fail("Invalid cast. The supplied value " + value + " did not pass the test '" + ts.Debug.getFunctionName(test) + "'.");
                        ^
Error: Debug Failure. Invalid cast. The supplied value [object Object] did not pass the test 'isOptionalChain'.
    at Object.cast (/home/hacksign/Code/blogstation/node_modules/typescript/lib/typescript.js:1693:25)
    at flattenChain (/home/hacksign/Code/blogstation/node_modules/typescript/lib/typescript.js:85256:28)
    at visitOptionalExpression (/home/hacksign/Code/blogstation/node_modules/typescript/lib/typescript.js:85311:22)
    at visitor (/home/hacksign/Code/blogstation/node_modules/typescript/lib/typescript.js:85236:39)
    at visitNode (/home/hacksign/Code/blogstation/node_modules/typescript/lib/typescript.js:78322:23)
    at Object.visitEachChild (/home/hacksign/Code/blogstation/node_modules/typescript/lib/typescript.js:78712:225)
    at visitor (/home/hacksign/Code/blogstation/node_modules/typescript/lib/typescript.js:85249:31)
    at visitNodes (/home/hacksign/Code/blogstation/node_modules/typescript/lib/typescript.js:78375:48)
    at Object.visitEachChild (/home/hacksign/Code/blogstation/node_modules/typescript/lib/typescript.js:78714:68)
    at visitor (/home/hacksign/Code/blogstation/node_modules/typescript/lib/typescript.js:85249:31)
```

webpack.config.js:

```
import path from 'path';
import { Program } from 'typescript';
import webpack from 'webpack';
import LoadablePlugin from '@loadable/webpack-plugin';
import { createLoadableComponentsTransformer } from 'typescript-loadable-components-plugin';

const config : webpack.Configuration = {
  mode: process.env.NODE_ENV === 'development' ? 'development' : 'production',
  target: 'web',
  devtool: 'source-map',
  entry: {
    index: path.resolve(__dirname, '../frontend/containers/default/index.tsx')
  },
  output: {
    filename: '[name].bundle.js',
    chunkFilename: '[name].chunk.js',
    path: path.resolve(__dirname, '../dist'),
    publicPath: '/',
  },
  optimization: {
    splitChunks: {
      chunks: 'all',
      cacheGroups: {
        vendor: {
          test: /[\\/]node_modules[\\/]/,
          name: (module) => {
            const packageName = module.context.match(/[\\/]node_modules[\\/](.*?)([\\/]|$)/)[1];
            return `npm.${packageName.replace('@', '')}`;
          },
        }
      }
    }
  },
  module: {
    rules: [
      {
        test: /\.tsx?$/,
        exclude: /node_modules/,
        loader: 'ts-loader',
        options: {
          getCustomTransformers: (program : any) => ({
            before: [createLoadableComponentsTransformer(program as Program, {})],
          }),
        },
      },
    ],
  },
  resolve: {
    extensions: [ '.tsx', '.ts', '.js' ],
  },
  plugins: [
    new LoadablePlugin(),
  ]
}

export default config;
```

And the file caused the problem:

```
import React from 'react';
import { Route } from 'react-router-dom';
import loadable from '@loadable/component';
import Header from '../../components/header';
import Poster from '../../components/poster';
import Footer from '../../components/footer';
import Content from '../../components/content';

const HeaderAsync = loadable(
  () => import('../../components/header')
)
const Component = () => {
  return (
    <React.Fragment>
        <HeaderAsync />  // ERROR HERER: will receive above error when use loadable, remove this component will pass the compile
        <Poster />
        <Route path='/' exact component={Content} />
        <Footer />
    </React.Fragment>
  );
}

export default Component;
```

**Comment by ibobo** [comment]
Same here! Did you find a solution?

**Comment by Hacksign** [comment]
> Same here! Did you find a solution?

change tool chain to babel + presets. Then use `@loadable/component` adviced configuration.

**Comment by Shvembldr** [comment]
Same problem

**Comment by PLSFIX** [comment]
Same here

**Comment by miniven** [comment]
Have same problem too

**Comment by acrazing** [comment]
This maybe fixed by #6. You can try it by upgrade `typescript-loadable-components-plugin` to the latest version `1.0.2`.

```bash
yarn add typescript-loadable-components-plugin@latest

# or npm
npm install -S typescript-loadable-components-plugin@latest
```
